src/components/PostReview.jsx

```diff
@@ -0,0 +1,136 @@
+import React, { useState, useEffect } from 'react';
+import { Modal, Button, Form } from 'react-bootstrap';
+import { useAuth } from "./AuthContext";
+import {
+  StarFill,
+  Star
+} from "react-bootstrap-icons";
+
+const PostReview = ({ id, show, onHide, reloadReviews }) => {
+  const [username, setUsername] = useState('');
+  const [review, setReview] = useState('');
+  const [difficulty, setDifficulty] = useState(1);
+  const [data2, setData2] = useState(null);
+  const [image, setImage] = useState(null);
+  const { token } = useAuth();
+
+  useEffect(() => {
+    fetch(process.env.REACT_APP_API_URL + "/user/me", {
+        method: "GET",
+        headers: {
+        Authorization: `Bearer ${token}`,
+        },
+    })
+        .then((response) => response.json())
+        .then((data) => {
+          setData2(data);
+          setUsername(data?.username);
+        })
+        .catch((error) => console.error("Error:", error));
+  }, [id])
+
+  const handleFileChange = (event) => {
+    const selectedImage= event.target.files[0];
+    setImage(selectedImage);
+  };
+
+  const handlePostReview = async () => {
```

Comment:
In this method we should control if the review is empty (the text) or not

Reply:
Or control it from the backend and launch a popup, but I think it is more feasible to control it in the frontend

Reply:
The text is optional, can be empty